lxd/instance_post.go

```diff
@@ -280,6 +297,80 @@ func instancePost(d *Daemon, r *http.Request) response.Response {
 	return operations.OperationResponse(op)
 }
 
+// Move an instance to another pool.
+func instancePostPoolMigration(d *Daemon, inst instance.Instance, newName string, instanceOnly bool, newPool string, op *operations.Operation) error {
+	if inst.IsRunning() {
+		return fmt.Errorf("Instance must not be running to move between pools")
+	}
+
+	if inst.IsSnapshot() {
+		return fmt.Errorf("Instance snapshots cannot be moved between pools")
+	}
+
+	// Copy config from instance to avoid modifying it.
+	localConfig := make(map[string]string)
+	for k, v := range inst.LocalConfig() {
+		localConfig[k] = v
+	}
+
+	// Load source root disk from expanded devices (in case instance doesn't have its own root disk).
+	rootDev := inst.ExpandedDevices()["root"]
+	rootDev["pool"] = newPool
+
+	// Copy device config from instance, and update target instance root disk device with the new pool name.
+	localDevices := inst.LocalDevices().Clone()
+	localDevices["root"] = rootDev
```

Comment:
We shouldn't assume that the root disk is called `root`.

Reply:
@stgraber true, fixed.